web/modules/custom/dpl_consumers/dpl_consumers.module

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/**
+ * @file
+ * DPL consumers module.
+ */
+
+/**
+ * Get consumer by Client ID.
+ *
+ * @return string
+ *   The consumer UUID.
+ */
+function dpl_consumers_get_consumer_uuid(string $client_id): string {
+  try {
+    $consumer = \Drupal::entityTypeManager()
+      ->getStorage('consumer')
+      ->loadByProperties(['client_id' => $client_id]);
+
+    if (!empty($consumer)) {
+      $consumer = reset($consumer);
+      $uuid = $consumer->uuid();
+      return $uuid ?? 'Consumer UUID not found.';
+    }
+    else {
+      return 'Consumer not found.';
+    }
```

Comment:
i dont like the behaviour of here. So the contract of the function is that it will always return a string. The string can either be:
* A uuid string
* "Consumer UUID not found."
I think the return should only be a uuid string. If you cannot resolve the uuid you should throw an exception.

Reply:
We now throw exceptions when we do not get the expected result.